dahee/services/userService.js

```diff
@@ -10,9 +14,31 @@ const signUp = async (name, email, password, profileImage) => {
     err.statusCode = 400;
     throw err;
   }
-  return userDao.createUser(name, email, password, profileImage);
+  const hashedPassword = await bcrypt.hash(password, saltRounds);
+  return userDao.createUser(name, email, hashedPassword, profileImage);
+};
+
+const signIn = async (email, password) => {
+  try {
+    const userObj = await userDao.verifyUser(email);
+    const authResult = await bcrypt.compare(password, userObj.password);
+    if (authResult) {
+      const payload = { userId: userObj.id };
+      const token = jwt.sign(payload, secretKey);
+      return token;
+    } else {
+      const error = new Error('INVALID_USER');
+      error.statusCode = 400;
+      throw error;
+    }
```

Comment:
1.  if의 조건에 에러가 발생하는 상황(비정상적인 상황)에 대한 조건을 넣어 조건을 만족했을 때 error가 나게하면 다음 로직에 대해 else 없이 코드를 진행 할 수 있어 더 읽기 좋은 코드가 됩니다.
2. auth 관련 error는 401 상태코드를 사용해 주세요.
```suggestion
    
    if (!authResult) {
      const error = new Error('INVALID_USER');
      error.statusCode = 401;
      throw error;
    }
    
    const payload = { userId: userObj.id };
    const token = jwt.sign(payload, secretKey);
    return token; 
```